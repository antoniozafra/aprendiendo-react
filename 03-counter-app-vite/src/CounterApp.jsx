
import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({ value }) => {

  const [counter, setCounter] = useState(value);


  const handdleAdd = (value) => {
    // setCounter((c) => c + 1);
    setCounter(counter + 1);

  }

  const handdleSubstract = () => {
    if (counter == value) {
      return
    }

    setCounter(counter - 1);
  }
  const handdleReset = () => {
    setCounter(value);
  }


  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>

      <button onClick={handdleAdd}>+1</button>
      <button onClick={handdleSubstract}>-1</button>
      <button onClick={handdleReset}>Reset</button>
    </>
  );
}

//Properties
CounterApp.propTypes = {
  value: PropTypes.number.isRequired

}