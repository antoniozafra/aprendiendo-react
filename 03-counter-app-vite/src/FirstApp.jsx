import Proptypes from 'prop-types';


export const FirstApp = ( {title, subtitle, name} ) => {

    return (
        //Return a Fragment
        <>
        
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    );
}



FirstApp.propTypes = {
    title : Proptypes.string.isRequired,
    // subtitle : Proptypes.number
}


FirstApp.defaultProps = {
    subtitle: ' No hay subtitutlo',
    name: ' Antonio z',
    // title : 'No hay titulo'
}

