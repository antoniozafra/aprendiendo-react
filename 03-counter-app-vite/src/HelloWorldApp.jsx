export const HelloWorldApp = () => {
  return (
    <div>
        <h1>hello world desde react</h1>
    </div>
  )
}

