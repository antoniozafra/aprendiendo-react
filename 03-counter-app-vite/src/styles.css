body {
    background-color: #21232a;
    color: white;
    font-size: 25px;
    padding: 70px;
}

body button {
    padding: 10px 20px;
    margin: 5px;
}