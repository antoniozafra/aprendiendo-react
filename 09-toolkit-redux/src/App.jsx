import { useDispatch, useSelector } from 'react-redux';
import './App.css'
import { decrement, increment, incrementBy } from './store/slices/counter';
function App() {
  const {counter} = useSelector( state => state.counter);

  const dispatch = useDispatch();


  return (
    <>
      <div>
      </div>
      <h1>Counter is {counter}</h1>
      <div >
        <button onClick={() => dispatch(increment())}>
          Increment
        </button>

        <button onClick={() => dispatch(decrement())}>
          Decrement
        </button>

        <button onClick={() => dispatch(incrementBy({increment: 2}))}>
          Increment by X
        </button>
        
      </div>
    </>
  )
}

export default App
