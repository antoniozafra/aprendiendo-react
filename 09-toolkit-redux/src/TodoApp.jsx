import React from 'react'
import { useGetTodosQuery } from './store/apis'

export const TodoApp = () => {

    const {data, isLoading, isFetching} = useGetTodosQuery();

  return (
    <>
        <h1>Todo rtk</h1>
        <hr />

        <h4>isLoading: {isLoading ? 'true' : 'false'}</h4>

        <pre>...</pre>
        <ul>
            {/* {data.map( todo => (
                <li key={todo.id}>{todo.id}</li>
            ))} */}
        </ul>

        <button>NextTodo</button>
    </>
  )
}
