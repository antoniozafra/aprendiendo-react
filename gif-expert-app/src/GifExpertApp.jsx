import  { useState } from 'react'
import {AddCategory, GifGrid} from './components';
// import GifGrid from './components/GifGrid';



export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Superman']);
   
    const onAddCategory = ( newCategory ) => {

        if(categories.includes(newCategory)) return;

        // setCategories([...categories, "Valorant"]);
        setCategories(cat => [ newCategory, ...cat]);                                           

    }
    
  return (
    <>
        <h1>GifExpertApp</h1> 

        <AddCategory
            onNewCategory={ value => onAddCategory(value) }
            //  setCategories={setCategories}
        />
        
        {
            categories.map( (category) =>(
                    <GifGrid key={category} category={category}/>  
                )
            )
        }
    </>
  )
}

export default GifExpertApp;
