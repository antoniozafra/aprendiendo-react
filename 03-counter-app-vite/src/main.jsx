import React, { StrictMode } from 'react';
import ReactDOM from 'react-dom/client';
import "./styles.css"
import {CounterApp} from './CounterApp';
import { FirstApp } from './FirstApp';


ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        {/* <CounterApp value={10} /> */}
        <FirstApp title ="Este es el titulio"/>
    </React.StrictMode>
);
