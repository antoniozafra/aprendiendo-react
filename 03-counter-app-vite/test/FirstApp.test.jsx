import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en el componente <FirstApp/>', () => {

    test('Debe de hacer match con el snapshot ', () => {

        const title = "Hola, soy Antonio";
        const {container} = render(<FirstApp title={title} />);

        expect(container).toMatchSnapshot();

    });

    test('Debe de mostar el titulo en h1', () => {

        const title = "Hola, soy Antonio";
        const {container,getByText} = render(<FirstApp title={title} />);
      
        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain(title);
    
    });
    


});
