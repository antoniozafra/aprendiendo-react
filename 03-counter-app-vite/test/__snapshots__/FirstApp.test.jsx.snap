// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <FirstApp/> Debe de hacer match con el snapshot  1`] = `
<div>
  <h1>
    Hola, soy Antonio
  </h1>
  <p>
     No hay subtitutlo
  </p>
  <p>
     Antonio z
  </p>
</div>
`;
